src/stelae/archive.rs

```diff
@@ -0,0 +1,188 @@
+            for (qualified_name, _) in dependencies.dependencies {
+                let parent_dir = self.path.clone();
+                let (org, name) = get_name_parts(&qualified_name)?;
+                if std::fs::metadata(parent_dir.join(&org).join(&name)).is_err() {
+                    // Stele does not exist on the filesystem, continue to traverse other Steles
+                    continue;
+                }
+                let child = Stele::new(
+                    &self.path,
+                    Some(name),
+                    Some(org.clone()),
+                    Some(parent_dir.join(org)),
+                    false,
+                )?;
+                self.stelae
+                    .entry(format!("{}/{}", child.auth_repo.org, child.auth_repo.name))
+                    .or_insert_with(|| child.clone());
+                self.traverse_children(&child)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+/// Check if the `path` is inside an existing archive
+/// # Errors
+/// Return an error if the path is inside an existing archive.
+fn raise_error_if_in_existing_archive(path: &Path) -> anyhow::Result<bool> {
+    let existing_archive_path = find_archive_path(path);
+    match existing_archive_path {
+        Ok(_) => anyhow::bail!("You cannot create a new archive inside of an existing archive."),
+        Err(_) => Ok(false),
+    }
+}
+
+/// Config object for an Archive
+#[derive(Deserialize, Serialize)]
+pub struct Config {
+    /// The root Stele for this archive
+    pub root: stele::Config,
+    /// Whether this is a shallow archive (all repos depth=1)
+    pub shallow: bool,
+    /// Custom HTTP headers used to interact with the Stele
+    pub headers: Option<Headers>,
+}
+
+/// Optional Header configuration for an Archive
+#[derive(Default, Deserialize, Serialize)]
+pub struct Headers {
+    /// Specify a custom header guard to use when requesting a Stele's current documents.
+    pub current_documents_guard: Option<String>,
+}
+
+/// Create a new Stelae Archive at path, and return the new archive.
+/// # Errors
+/// Will error if archive is created inside of an existing archive.
+pub fn init(
+    path: PathBuf,
+    root_name: String,
+    root_org: String,
+    root_hash: Option<String>,
+    shallow: bool,
+    headers: Option<Headers>,
+) -> anyhow::Result<Box<Archive>> {
+    raise_error_if_in_existing_archive(&path)?;
+    let stelae_dir = path.join(PathBuf::from("./.stelae"));
+    create_dir_all(&stelae_dir)?;
+    let config_path = stelae_dir.join(PathBuf::from("./config.toml"));
+    let conf = Config {
+        root: stele::Config {
+            name: root_name,
+            org: root_org,
+            hash: root_hash,
+        },
+        shallow,
+        headers,
+    };
+    let conf_str = toml_edit::ser::to_string_pretty(&conf)?;
+    write(config_path, conf_str)?;
+    let archive = Archive {
+        path,
+        stelae: HashMap::new(),
+    };
+    Ok(Box::new(archive))
```

Comment:
Yes, although the function isn't used at the moment other than in the testing framework, so it might be unnecessary to Box it. But I'd say we leave the current Box for now